src/telemetry/otel.py

```
@@ -0,0 +1,192 @@
+    tracer_provider = TracerProvider(resource=resource)
+    if otlp_endpoint:
+        try:
+            from opentelemetry.exporter.otlp.proto.grpc.trace_exporter import (
+                OTLPSpanExporter,
+            )
+
+            tracer_provider.add_span_processor(
+                BatchSpanProcessor(OTLPSpanExporter(endpoint=otlp_endpoint))
+            )
+            logger.info("OTel traces → OTLP gRPC at %s", otlp_endpoint)
+        except Exception:
```

Ruff (0.16.0)

[warning] Do not catch blind exception: `Exception`

(BLE001)